.github/workflows/index-resources.yml

```diff
@@ -10,6 +10,13 @@ on:
   # Manually triggered using GitHub's UI
   workflow_dispatch:
 
+  workflow_call:
+    inputs:
+      ref:
+        description: Ref of the repo to use to generate the resource index.
+        type: string
+        required: true
```

Comment:
With automated calls happening more frequently than once a day or an ad-hoc, manual basis, I think we want to add a `concurrency` restriction to this workflow (or maybe just the job?) so that multiple invocations aren't stepping on each other's toes.

Reply:
Good idea! I'll probably use the [concurrency groups](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-concurrency-groups) to restrict by `github.head_ref` since each PR will be uploading different to different tmp S3 paths.

Reply:
Added concurrency restriction in https://github.com/nextstrain/nextstrain.org/pull/841/commits/33e8b990b51bd096d27d5cb14305e8b4c15c78b4, but I'm not entirely happy with it since different branches can still be using the same RESOURCE_INDEX. 

Hmm, I guess move the `aws cp` to a separate job and put a concurrency restriction on that job based on the `RESOURCE_INDEX` value.

Reply:
Nod. Restricting [`concurrency` at the job level](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency) by `RESOURCE_INDEX` could work without the separate job if the [job matrix grew to include `RESOURCE_INDEX` too](https://github.com/nextstrain/nextstrain.org/pull/841#discussion_r1599128769) instead of just `ref`.

Reply:
Updated concurrency to job level with `RESOURCE_INDEX` in https://github.com/nextstrain/nextstrain.org/pull/841/commits/4da1e8c8340a379b8a32daeaa65a80834482b6f1